x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/ltr/LearningToRankRescorerFeature.java

```diff
@@ -17,12 +19,22 @@
  * See {@link LearningToRankRescorerBuilder}
  */
 public class LearningToRankRescorerFeature {
+    /**
+     * This setting behave like a feature flag for the learning to rank feature but can be set by an operator on self-managed environment.
+     * The feature is enabled by default only for snapshots builds.
+     *
+     * See {@link LearningToRankRescorerBuilder}
+     */
+    public static Setting<Boolean> LEARNING_TO_RANK_ENABLED = Setting.boolSetting(
+        "xpack.ml.learning_to_rank.enabled",
+        Build.current().isSnapshot(),
+        Setting.Property.OperatorDynamic,
```

Comment:
If this is what we are going to do, it shouldn't be dynamic.

Reply:
Agreed!
I will remove the OperatorDynamic prop.